src/bindgen/language_backend/java_jna.rs

```diff
@@ -0,0 +1,860 @@
+use crate::bindgen::ir::{
+    Constant, Documentation, Enum, Field, Function, IntKind, Item, Literal, OpaqueItem,
+    PrimitiveType, Static, Struct, Type, Typedef, Union,
+};
+use crate::bindgen::language_backend::LanguageBackend;
+use crate::bindgen::writer::ListType::Join;
+use crate::bindgen::writer::SourceWriter;
+use crate::bindgen::{Config, Layout};
+use std::fmt::Debug;
+use std::io::Write;
+
+pub struct JavaJnaLanguageBackend<'a> {
+    config: &'a Config,
+    binding_lib_crate_name: String,
+}
+
+impl<'a> JavaJnaLanguageBackend<'a> {
+    pub fn new(config: &'a Config, binding_lib_crate_name: String) -> Self {
+        Self {
+            config,
+            binding_lib_crate_name,
+        }
+    }
+}
+
+impl LanguageBackend for JavaJnaLanguageBackend<'_> {
+    fn open_namespaces<W: Write>(&mut self, out: &mut SourceWriter<W>) {
+        out.new_line_if_not_start();
+        let name = &self
+            .config
+            .java_jna
+            .interface_name
+            .clone()
+            .unwrap_or("Bindings".to_string());
```

Comment:
no need to clone / allocate here, right?

Doesn't: `self.config.java_jna.interface_name.as_deref().unwrap_or("Bindings")` work?

Reply:
:+1: